pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1248,3 +1248,48 @@ func TestGetDeploymentsForReplicaSet(t *testing.T) {
 	}
 
 }
+func generateRSWithAnnounces(revision string) apps.ReplicaSet {
+	return apps.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:             randomUID(),
+			Name:            names.SimpleNameGenerator.GenerateName("replicaset"),
+			Labels:          map[string]string{"foo": "bar"},
+			Annotations:     map[string]string{RevisionAnnotation: revision},
+		},
+	}
+}
+
+func TestLastRevision(t *testing.T) {
+	rs1 := generateRSWithAnnounces("1")
+	rs2 := generateRSWithAnnounces("2")
+	rs3 := generateRSWithAnnounces("3")
+	RSs1 := []*apps.ReplicaSet{&rs1,}
+	RSs2 := []*apps.ReplicaSet{&rs1,&rs2,}
+	RSs3 := []*apps.ReplicaSet{&rs1,&rs2,&rs3}
+
+	tests := []struct {
+		RSs             []*apps.ReplicaSet
+		expectRevision int64
+	}{
+		{
+			RSs: RSs1,
+			expectRevision: 0,
+		},
+		{
+			RSs: RSs2,
+			expectRevision: 1,
+		},
+		{
+			RSs: RSs3,
+			expectRevision: 2,
+		},
+	}
+	for _, test := range tests {
+		RSs := test.RSs
+		res := LastRevision(RSs)
```

Comment:
```suggestion
		res := LastRevision(test.replicaSets)
```

Although it's more common to use `tc` instead of `test` in the codebase.